apis/v1beta1/httproute_types.go

```diff
@@ -303,6 +317,18 @@ const (
 )
 
 // HTTPPathMatch describes how to select a HTTP route by matching the HTTP request path.
+//
+// +kubebuilder:validation:XValidation:message="value must be an absolute path and start with '/' when type one of ['Exact', 'PathPrefix']",rule="(self.type == 'Exact' || self.type == 'PathPrefix') ? self.value.startsWith('/') : true"
+// +kubebuilder:validation:XValidation:message="must not contain '//' when type one of ['Exact', 'PathPrefix']",rule="(self.type == 'Exact' || self.type == 'PathPrefix') ? !self.value.contains('//') : true"
+// +kubebuilder:validation:XValidation:message="must not contain '/./' when type one of ['Exact', 'PathPrefix']",rule="(self.type == 'Exact' || self.type == 'PathPrefix') ? !self.value.contains('/./') : true"
+// +kubebuilder:validation:XValidation:message="must not contain '/../' when type one of ['Exact', 'PathPrefix']",rule="(self.type == 'Exact' || self.type == 'PathPrefix') ? !self.value.contains('/../') : true"
+// +kubebuilder:validation:XValidation:message="must not contain '%2f' when type one of ['Exact', 'PathPrefix']",rule="(self.type == 'Exact' || self.type == 'PathPrefix') ? !self.value.contains('%2f') : true"
```

Comment:
no way to do caseless compare?  I guess you want to get the match value out, sort of like a regex...

Reply:
Yeah I think these are all split up the way that they are just to provide more meaningful error messages

Reply:
Thanks Joe! Would you agree that an expression like:

```
!(self.type in ['Exact', 'PathPrefix']) || !self.value.contains('%2f')

v/s

(self.type in ['Exact','PathPrefix']) ? !self.value.contains('%2f') : true
```

although definitely (a bit) shorter, sacrifices the comprehensibility of the conditional here? (I think just using the `in` seems good enough)